<a href="https://colab.research.google.com/github/Viswonathan06/Titanic-Dataset/blob/main/Titanic_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Titanic Dataset**

### **Using PANDAS and MATPLOTLIB to evaluate and figure out the specifics of the Titanic dataset and draw some obvious conclusions dervied from the result of our code**.




In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
df = pd.read_csv("https://gist.githubusercontent.com/michhar/2dfd2de0d4f8727f873422c5d959fff5/raw/fa71405126017e6a37bea592440b4bee94bf7b9e/titanic.csv")

# **Basic overview of the data**


---


First we take a look at the head, i.e the first five rows of data. This gives us the basic overview and tells us the different columns that are present.

The [x:y] function gives us values from x to y-1.



---

Information not obvious:

* survival - Survival (0 = No; 1 = Yes)
* class - Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd)
* Name - Name
* sex - Sex
* age - Age
* sibsp - Number of Siblings/Spouses Aboard
* parch - Number of Parents/Children Aboard
* ticket - Ticket Number
* fare - Passenger Fare
* cabin - Cabin
* embarked - Port of Embarkation (C = Cherbourg; Q = Queenstown; S = Southampton)


In [ ]:
df.head()

#can also be done by 
#df[0:5]



PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

# Beginning our analysis

Let's start off small and get a basic statistical overview. (some values here may or may not make sense, but let's see anyway.)

In [ ]:
df.describe()

PassengerId    Survived      Pclass  ...       SibSp       Parch        Fare
count   891.000000  891.000000  891.000000  ...  891.000000  891.000000  891.000000
mean    446.000000    0.383838    2.308642  ...    0.523008    0.381594   32.204208
std     257.353842    0.486592    0.836071  ...    1.102743    0.806057   49.693429
min       1.000000    0.000000    1.000000  ...    0.000000    0.000000    0.000000
25%     223.500000    0.000000    2.000000  ...    0.000000    0.000000    7.910400
50%     446.000000    0.000000    3.000000  ...    0.000000    0.000000   14.454200
75%     668.500000    1.000000    3.000000  ...    1.000000    0.000000   31.000000
max     891.000000    1.000000    3.000000  ...    8.000000    6.000000  512.329200

[8 rows x 7 columns]

We see that there are a total of 891 passengers taken into account in this database. The mean for the columns PassengerId, Pclass make no sense, so we shall ignore those. 

* We see that the mean of age is about 29.7 years, which means most of the people on board were middle-aged adults. 

* Mean fare was \$32.20 which means that there were a lot of people with tickets in the lower class comparatively. Maximum fare was \$512.33.

Rest of the data will be made clear as we move along.


Now let's take a look at the number of people who survived the sinking of Titanic.


In [ ]:
df.groupby('Survived').count()[['PassengerId']]

PassengerId
Survived             
0                 549
1                 342

Here we see that, the number of people who survived are 549 and number of peeople who died are 342.


---

Now let's analyse the following:
Amongst the number of survivors,
* how many had family on board and,
* what class in the boat did they reside.


---

First let's take a look at how many people were there in the various classes of the boat.





In [ ]:
df.groupby('Pclass').count()[['PassengerId']]

PassengerId
Pclass             
1               216
2               184
3               491

We notice that maximum people resided in third class as suspected above by looking at the mean fare


---

Now lets take a look at the survival numbers based on class


In [ ]:
df.groupby(['Survived', 'Pclass']).count()[['PassengerId']]

PassengerId
Survived Pclass             
0        1                80
         2                97
         3               372
1        1               136
         2                87
         3               119

Here we see that, amongst the people that **did not** survive, 80 were in the 1st class while 372 were in the third class. 

We notice that amongst the people in third class, with total of 491 people, 372 were dead and only 119 were alive. 

It was not the case in first class however, where 80 out of 216 died and 136 people were alive. 

Let's take a look at the number of male and female passengers.

In [ ]:
df.groupby('Sex').count()[['PassengerId']]

PassengerId
Sex                
female          314
male            577



Now let's look at the male to female ratio of the survivors and dead people.

In [ ]:
df.groupby(['Survived', 'Sex']).count()[['PassengerId']]

PassengerId
Survived Sex                
0        female           81
         male            468
1        female          233
         male            109

As we know that the women and children were evacuated first, the number of male deaths were staggeringly high as compared to the female. As a result, number of male survivers also were less compared to that over female. 

* 468/577 dead males
* 81/314 dead females

Let's analyse how many people were under the age of 18.

In [ ]:
df[df.Age <= 18].count()[['PassengerId']]

PassengerId    139
dtype: int64

139 people were under or of the age 18 i.e children. 


---

How many of these children survived the sinking?

In [ ]:
df[df.Age <= 18].groupby('Survived').count()[['PassengerId']]

PassengerId
Survived             
0                  69
1                  70

Sadly, about half the children did not make it out of the boat. 



---

Youngest person to die?


In [ ]:
#df.sort_values(by = 'Age',ascending= True)[['Survived','Age']]
dead = df[df.Survived == 0].sort_values(by = 'Age')
dead = dead[dead['Age'].notnull()] ## Here we choose the rows which dont have age as NULL to check the eldest recorded person. 
dead
# dead = df[df.Survived == 0]
# dead[dead.Age.min()]

PassengerId  Survived  Pclass  ...      Fare    Cabin  Embarked
164          165         0       3  ...   39.6875      NaN         S
386          387         0       3  ...   46.9000      NaN         S
642          643         0       3  ...   27.9000      NaN         S
205          206         0       3  ...   10.4625       G6         S
297          298         0       1  ...  151.5500  C22 C26         S
..           ...       ...     ...  ...       ...      ...       ...
672          673         0       2  ...   10.5000      NaN         S
116          117         0       3  ...    7.7500      NaN         Q
493          494         0       1  ...   49.5042      NaN         C
96            97         0       1  ...   34.6542       A5         C
851          852         0       3  ...    7.7750      NaN         S

[424 rows x 12 columns]

Here we see that, according to this dataset, youngest person to die on the ship was 1 year old. and eldest person to die, that we have a record of, is 74 years old. 



---

---

We shall now see some graphical representations of the above data using matplotlib. 


